rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java

```diff
@@ -320,85 +377,47 @@ public void accept(long n) {
-                            // Need to set this to one for first the frame
-                            wrapped.request(1);
-                            senders.put(streamId, wrapped);
-                            receivers.put(streamId, receiver);
-
-                            return wrapped;
-                          })
-                      .subscribe(
-                          new BaseSubscriber<Payload>() {
-
-                            boolean firstPayload = true;
-
-                            @Override
-                            protected void hookOnNext(Payload payload) {
-                              final ByteBuf frame;
-
-                              if (firstPayload) {
-                                firstPayload = false;
-                                frame =
-                                    RequestChannelFrameFlyweight.encode(
-                                        allocator,
-                                        streamId,
-                                        false,
-                                        false,
-                                        n,
-                                        payload.sliceMetadata().retain(),
-                                        payload.sliceData().retain());
-                              } else {
-                                frame =
-                                    PayloadFrameFlyweight.encode(
-                                        allocator, streamId, false, false, true, payload);
-                              }
-
-                              sendProcessor.onNext(frame);
-                              payload.release();
-                            }
-
-                            @Override
-                            protected void hookOnComplete() {
-                              if (contains(streamId) && !receiver.isDisposed()) {
-                                sendProcessor.onNext(
-                                    PayloadFrameFlyweight.encodeComplete(allocator, streamId));
-                              }
-                              if (firstPayload) {
-                                receiver.onComplete();
-                              }
-                            }
-
-                            @Override
-                            protected void hookOnError(Throwable t) {
-                              errorConsumer.accept(t);
-                              receiver.dispose();
-                            }
-                          });
-                } else {
-                  if (contains(streamId) && !receiver.isDisposed()) {
-                    sendProcessor.onNext(RequestNFrameFlyweight.encode(allocator, streamId, n));
+                  senders.put(streamId, upstreamSubscriber);
+                  receivers.put(streamId, receiver);
+
+                  inboundFlux.limitRate(Queues.SMALL_BUFFER_SIZE).subscribe(upstreamSubscriber);
```

Comment:
@rstoyanchev for your attention. I got to read of the custom rate limiter and use a built-in one